seed/views/portfoliomanager.py

```diff
@@ -234,6 +265,33 @@ def generate_and_download_template_report(self, matched_template):
             response = requests.get(download_url, headers=self.authenticated_headers)
         except requests.exceptions.SSLError:
             raise PMExcept("SSL Error in Portfolio Manager Query; check VPN/Network/Proxy.")
-        if not response.status_code == 200:
+        if not response.status_code == status.HTTP_200_OK:
+            raise PMExcept("Unsuccessful response from GET trying to download generated report; aborting.")
+        return response.content
+
+    def generate_and_download_child_data_request_report(self, matched_data_request):
```

Comment:
This new worker function simply makes one URL request to get the XML data response.